src/keyring/KeyringController.ts

```diff
@@ -577,6 +596,94 @@ export class KeyringController extends BaseController<
     this.update({ keyrings: [...keyrings] });
     return privates.get(this).keyring.fullUpdate();
   }
+
+  // QR Hardware related methods
+
+  /**
+   * Add qr hardware keyring.
+   *
+   * @returns The added keyring
+   */
+  private async addQRKeyring() {
+    return await privates.get(this).keyring.addNewKeyring(KeyringTypes.qr);
+  }
+
+  /**
+   * Get qr hardware keyring.
+   *
+   * @returns The added keyring
+   */
+  private async getQRKeyring() {
+    const keyring = privates
+      .get(this)
+      .keyring.getKeyringsByType(KeyringTypes.qr)[0];
+    if (keyring) {
+      return keyring;
+    }
+    return await this.addQRKeyring();
+  }
+
+  getQRKeyringState = async () => {
+    return (await this.getQRKeyring()).getMemStore();
+  };
+
+  submitQRKeyring = async (cryptoHDKey: any) =>
+    (await this.getQRKeyring()).syncKeyring(cryptoHDKey);
+
+  submitQRCryptoHDKey = async (cryptoHDKey: any) =>
+    (await this.getQRKeyring()).submitCryptoHDKey(cryptoHDKey);
+
+  cancelSyncQRCryptoHDKey = async () =>
+    // eslint-disable-next-line node/no-sync
+    (await this.getQRKeyring()).cancelSync();
+
+  submitQRHardwareSignature = async (requestId: string, ethSignature: any) =>
+    (await this.getQRKeyring()).submitSignature(requestId, ethSignature);
+
+  cancelQRHardwareSignRequest = async () =>
+    (await this.getQRKeyring()).cancelSignRequest();
+
+  connectQRHardware = async (page: number) => {
+    const keyring = await this.getQRKeyring();
+    let accounts = [];
+    switch (page) {
+      case -1:
+        accounts = await keyring.getPreviousPage();
+        break;
+      case 1:
+        accounts = await keyring.getNextPage();
+        break;
```

Comment:
Good catch, I will add a check here.

Reply:
Hi @gantunesr , I checked the code again and if there has no previous page, it will return the first page of accounts. So I add an try/catch to catch the error about exceeding the boundary for `next page`